tests/chainer_tests/test_variable.py

```diff
@@ -983,358 +977,314 @@ def v():
-                backend.CpuDevice().send(x_var.grad),
-                backend.CpuDevice().send(gx))
-            assert x_var.grad_var is not None
-            assert x_var.grad_var.node is not None
-        else:
-            assert x_var.grad is None
-            assert x_var.grad_var is None
-
-        orig_device = backend.get_device_from_array(x)
-        if orig_device == expected_device:
-            assert x_var.data is x
-            assert x_var.grad is gx
+                backend.CpuDevice().send(x.grad),
+                backend.CpuDevice().send(gx_src))
         else:
-            assert x_var.data is not x
-            assert not set_grad_var or x_var.grad is not gx
+            assert x.grad is None
+            assert x.grad_var is None
 
-        assert x_var.xp is not chainerx
+    def test_to_device_meth_allow_unchain(
+            self, src_backend_config, dst_backend_config):
+        self.setup(src_backend_config)
 
-    def test_to_gpu_from_cpu(self):
-        self.check_to_gpu(self.x, self.gx)
+        self.to_device_meth(dst_backend_config.device, True)
 
-    def test_to_gpu_from_gpu(self):
-        self.check_to_gpu(cuda.to_gpu(self.x), cuda.to_gpu(self.gx))
+        self.check_to_device_meth_devices(
+            src_backend_config.device, dst_backend_config.device)
+        self.check_to_device_meth_ndarrays()
 
-    @attr.multi_gpu(2)
-    def test_to_gpu_from_another_gpu(self):
-        self.check_to_gpu(cuda.to_gpu(self.x), cuda.to_gpu(self.gx), 1)
+    def test_to_device_meth_not_allow_unchain(
+            self, src_backend_config, dst_backend_config):
+        self.setup(src_backend_config)
 
-    @attr.chainerx
-    def test_to_gpu_from_chainerx(self):
-        self.check_to_gpu(
-            chainerx.array(self.x),
-            chainerx.array(self.gx),
-            requires_grad=False)
+        if self.x_with_graph:
+            with pytest.warns(DeprecationWarning):
+                self.to_device_meth(dst_backend_config.device, False)
+        else:
+            # Check that nothing is warned.
+            self.to_device_meth(dst_backend_config.device, False)
 
-    @attr.chainerx
-    def test_to_gpu_from_chainerx_requiring_grad(self):
-        with self.assertRaises(RuntimeError):
-            self.check_to_gpu(
-                chainerx.array(self.x),
-                chainerx.array(self.gx),
-                requires_grad=True)
 
+@testing.parameterize(*testing.product({
+    'x_shape': [(10,), ()],
+    'x_with_grad': [True, False],
+    'x_with_graph': [True, False],
+}))
+@testing.backend.inject_backend_tests(None, _numpy_backend_params)
+@testing.backend.inject_backend_tests(None, _backend_params)
+class TestVariableToCpu(VariableToDeviceFamilyBase, unittest.TestCase):
 
-@testing.parameterize(
-    {'x_shape': (10,)},
-    {'x_shape': ()},
-)
-@attr.chainerx
-class TestVariableToChainerX(unittest.TestCase):
+    def to_device_meth(self, _, allow_unchain):
```

Comment:
Let's not replace the unused argument with `_`.
It's not clear what it was and that it's really something to ignore.
Ditto for others.

Reply:
Fixed!